enumerables.rb

```diff
@@ -0,0 +1,270 @@
+      if self.class == Array or self.class == Range
+        my_each do |elmt|
+          new_arr.push(yield(elmt))
+        end
+      elsif self.class == Hash
+        my_each do |key, value|
+          new_arr.push(yield(key, value))
+        end
+      end
+      new_arr
+
+    elsif my_proc[0].class == Proc or (my_proc[0].class == Proc && block_given?) # Case 2: When the method takes a proc
+      if self.class == Array or self.class == Range
+        my_each do |i|
+          new_arr.push(my_proc[0].call(i))
+        end
+      elsif self.class == Hash
+        my_each do |key, value|
+          new_arr.push(my_proc[0].call(key, value))
+        end
+      end
+      new_arr
+
+    elsif my_proc.empty? && !block_given? # Case 3: When there's no block or proc, the method returns enumerator
+      to_enum(:my_map)
+    end
+  end
+
+  # 9.- ---- my_inject Method ----
+  def my_inject(*args)
+    memo = to_a[0]
+
+    # Case 1: arguments are given but not a block
+    if !args.empty? && !block_given?
+
+      # There are an initial value and a symbol which indicates the operation to perform
+      if (args[0].is_a? Numeric or args[0].class == String) && args[1].class == Symbol
+        memo = args[0]
+        if args[1] == :+
+          my_each { |elmt| memo += elmt }
+        elsif args[1] == :-
+          my_each { |elmt| memo -= elmt }
+        elsif args[1] == :*
+          my_each { |elmt| memo *= elmt }
+        elsif args[1] == :/
+          my_each { |elmt| memo /= elmt }
+        elsif args[1] == :**
+          my_each { |elmt| memo **= elmt }
+        elsif args[1] == :%
+          my_each { |elmt| memo %= elmt }
+        end
+      elsif args.size == 1 && args[0].class == Symbol # There's only the operation's symbol
+        if args[0] == :+
+          to_a[1..-1].my_each { |elmt| memo += elmt }
+        elsif args[0] == :-
+          to_a[1..-1].my_each { |elmt| memo -= elmt }
+        elsif args[0] == :*
+          to_a[1..-1].my_each { |elmt| memo *= elmt }
+        elsif args[0] == :/
+          to_a[1..-1].my_each { |elmt| memo /= elmt }
+        elsif args[0] == :**
+          to_a[1..-1].my_each { |elmt| memo **= elmt }
+        elsif args[0] == :%
+          to_a[1..-1].my_each { |elmt| memo %= elmt }
+        end
+      end
+      memo
+
+      # Case 2: an initial value is given as an argument along with a block
+    elsif (args[0].is_a? Numeric or args[0].class == String) && block_given?
+      memo = args[0]
+      my_each { |elmt| memo = yield(memo, elmt) }
+      memo
+
+      # Case 3: only a block is given
+    elsif args.empty? && block_given?
+      to_a[1..-1].my_each { |elmt| memo = yield(memo, elmt) }
+      memo
+
+    elsif args.empty? && !block_given?
+      call_block = -> { yield 42 }
+      call_block.call
+
+    end
+  end
+
+  # 10.- ---- Method created for testing my_inject method ----
+
+  def multiply_els(array)
+    array.my_inject(:*)
+  end
```

Comment:
- [ ] `multiply_els` works fine, but it's supposed to be outside the module, take it out and that will be all :)